core/shared/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -137,6 +142,14 @@ object StaticFile {
     })
   }
 
+  private def calcETagURL[F[_]](implicit F: Sync[F]): URL => F[ETag] = url => {
+    val urlConn = url.openConnection
+    for {
+      lastModified <- F.blocking(urlConn.getLastModified.toHexString)
+      contentLength <- F.blocking(urlConn.getContentLengthLong.toHexString)
```

Comment:
`getLastModified` returns `0` if unknown and `getContentLengthLong` returns `-1`. Should we be doing something special in these situations? Otherwise seems like the calculated etag could be misleading.

Reply:
Note that in existing code we don't use blocking around lastModified/getContentLengthLong or openConnection.  Should those be changed?

Reply:
Changed to return Option if both are empty.